storage/src/key_value/agent/block.rs

```diff
@@ -223,34 +227,49 @@ impl<S: KeyValueStorage + 'static> Agent<S> {
         })
     }
 
-    pub(super) fn get_block_locator_hashes(&mut self) -> Result<Vec<Digest>> {
+    pub(super) fn get_block_locator_hashes(
+        &mut self,
+        points_of_interest: Vec<Digest>,
+        oldest_fork_threshold: usize,
+    ) -> Result<Vec<Digest>> {
         let canon = self.canon()?;
+        let target_height = canon.block_height as u32;
 
         // The number of locator hashes left to obtain; accounts for the genesis block.
-        let mut num_locator_hashes = std::cmp::min(crate::NUM_LOCATOR_HASHES - 1, canon.block_height as u32);
+        let mut num_locator_hashes = std::cmp::min(crate::NUM_LOCATOR_HASHES - 1, target_height);
```

Comment:
if the final number of locator hashes is greater than `NUM_LOCATOR_HASHES`, the `snarkos_network::MAX_BLOCK_SYNC_COUNT` value needs to be adjusted, so that more hashes can be processed - otherwise, the hashes above `NUM_LOCATOR_HASHES` would be ignored (it was an anti-DoS measure, so the node can't be spammed with too many locator hashes.